pkg/avx/collector.go

```diff
@@ -1,22 +1,247 @@
 package avx
 
 import (
-	"debug/elf"
-	"encoding/binary"
-	"flag"
+	"bytes"
 	"fmt"
-	"path"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
 	"sync"
 	"syscall"
 	"unsafe"
 
+	bpf "github.com/cilium/ebpf"
 	"github.com/intel/cri-resource-manager/pkg/cgroups"
 	logger "github.com/intel/cri-resource-manager/pkg/log"
-	bpf "github.com/iovisor/gobpf/elf"
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sys/unix"
 )
 
+var program = [...]byte{
```

Comment:
Wow. How was this generated and where is its source ?

Reply:
it's from: `xxd -i libexec/avx512.o`

Reply:
Ah, ok, it's just that. I thought it was something generated by some Cilium-specific toolchain.

Reply:
not yet but there's one and I'm planning to see if we could move to that.